src/time-lru-set.js

```diff
@@ -0,0 +1,164 @@
+//
+// Copyright 2019 DxOS.
+//
+
+class ItemCache {
+  constructor ({ value, maxAge, onTimeout }) {
+    this._value = value;
+    this._maxAge = maxAge;
+    this._onTimeout = value => onTimeout(value);
+  }
+
+  update () {
+    this.clear();
+
+    this._timer = setTimeout(() => {
+      this._onTimeout(this._value);
+    }, this._maxAge);
+  }
+
+  clear () {
+    if (this._timer) {
+      clearTimeout(this._timer);
+      this._timer = null;
+    }
+  }
+}
+
+/**
+ * Least Recently Used cache "set" structure with time live support.
+ *
+ */
+class TimeLRUSet {
+  /**
+   * @param {Object} options
+   * @param {number} [options.maxAge=10000] Maximum live time for a value not updated recently.
+   * @param {number} [options.maxSize=100] Max size of the cache.
+   */
+  constructor (options = {}) {
+    const { maxAge = 10 * 1000, maxSize = 100 } = options;
+
+    this._maxAge = maxAge;
+    this._maxSize = maxSize;
+
+    this._list = [];
+    this._map = new Map();
+  }
+
+  /**
+   * @type {number}
+   */
+  get size () {
+    return this._map.size;
+  }
+
+  /**
+   * Method returns a new Iterator object that contains the values
+   * for each element in the Set object in insertion order.
+   *
+   * @returns {Iterator<*>}
+   */
+  values () {
+    return this._map.keys();
+  }
+
+  /**
+   * Add a value.
+   *
+   * @param {*} value Value to store.
+   * @returns {TimeLRUSet}
+   */
+  add (value) {
+    if (this.has(value)) {
+      return this;
+    }
+
+    if (this._list.length === this._maxSize) {
+      this._deleteLRU();
+    }
+
+    const item = new ItemCache({
+      value,
+      maxAge: this._maxAge,
+      onTimeout: this._onTimeout.bind(this)
```

Comment:
Just inline the method here. {value => this.delete(value)}

Reply:
fixed